LOCAL_VIRTUALENV.rst

```diff
@@ -18,115 +18,156 @@
 
+You are STRONGLY encouraged to also install and use `pre-commit hooks <CONTRIBUTING.rst#pre-commit-hooks>`_ 
+for your local virtualenv development environment. Pre-commit hooks can speed up your 
+development cycle a lot.
 
-Creating virtualenv can be automated with `Breeze environment <BREEZE.rst#configuring-local-virtualenv>`_
+The full list of extras is available in `<setup.py>`_.
 
-Once initialization is done, you should select the virtualenv you
-initialized as the project's default virtualenv in your IDE.
+Creating a Local virtualenv
+===========================
 
-After setting it up - you can use the usual "Run Test" option of the IDE
-and have the autocomplete and documentation support from IDE as well as
-you can debug and view the sources of Airflow - which is very helpful
-during development.
+To use your IDE for Airflow development and testing, you need to configure a virtual 
+environment. Ideally you should set up virtualenv for all Python versions that Airflow
+supports (3.5, 3.6). 
 
-Installing other extras
------------------------
+Consider using the
+`virtualenvwrapper <https://virtualenvwrapper.readthedocs.io/en/latest/>`__
+that allows you to create virtual environments and easily switch between them with the 
+``workon`` command. 
 
-You can also other extras (like ``[mysql]``, ``[gcp]`` etc. via
-``pip install -e [EXTRA1,EXTRA2 ...]``. However some of the extras have additional
-system requirements and you might need to install additional packages on your
-local machine.
+1. Create an environment as follows:
 
-For example if you have trouble installing mysql client on MacOS and you have
-an error similar to
+   ``mkvirtualenv <ENV_NAME> --python=python<VERSION>``
 
-.. code:: text
+2. Install Python PIP requirements:
 
-    ld: library not found for -lssl
+   ``pip install -e ".[devel]"``
 
-you should set LIBRARY\_PATH before running ``pip install``:
+3. Create the Airflow sqlite database:
 
-.. code:: bash
+   ``airflow db init``
 
-    export LIBRARY_PATH=$LIBRARY_PATH:/usr/local/opt/openssl/lib/
+4. Select the virtualenv you created as the project's default virtualenv in your IDE.
+
+Note that if you have the Breeze development environment installed, the ``breeze`` 
+script can _`automate creating a virtualenv <BREEZE.rst#using-your-host-ide-with-breeze>`_ 
+(steps 2 and 3).
 
-The full list of extras is available in `<setup.py>`_
 
+Debugging and Running Tests
+===========================
 
-Running individual tests
+Local and Remote Debugging
+--------------------------
+
+One of the great benefits of using the local virtualenv is an option to run
+local debugging in your IDE graphical interface. You can also use ``ipdb``
+if you prefer _`console debugging <BREEZE.rst#debugging-with-ipdb>`_.
+
+You can also benefit from the _`remote debugging <BREEZE.rst#debugging-airflow-breeze-tests-in-ide>`_
+option available in IntelliJ/PyCharm (paid version) and debug example DAGs 
+running in a separate container.
```

Comment:
I'd like to move the section BREEZE.rst#debugging-airflow-breeze-tests-in-ide to this doc. Do you think it makes sense?

Reply:
Yeah. That should be OK. However just thinking maybe we should have a separate TESTING.rst for example that we could describe both - testing with local virtualenv and Breeze? Just a thought - I do not feel strong about it, but I think since we already have few documents, it might make sense ? What do you think? I think you mentioned once that it's better to have one doc rather than many, but maybe in this case this will be justified since we can have links between the docs.

Reply:
Let's beef them up first and see whether more granular docs are required. Actually, adding a TOC pane would help to navigate in a bigger doc...